wsmaster/che-core-api-factory/src/main/java/org/eclipse/che/api/factory/server/FactoryManager.java

```diff
@@ -133,8 +141,12 @@ public Factory updateFactory(Factory update, Set<FactoryImage> images) throws Co
                                                                                   ServerException {
         requireNonNull(update);
         final AuthorImpl creator = factoryDao.getById(update.getId()).getCreator();
+        FactoryImpl updateImpl = new FactoryImpl(update, images);
+        if (isNullOrEmpty(updateImpl.getName())) {
```

Comment:
It would be better not to generate new name when update doesn't contain it. Let's set old name when it is null or empty in update.